services/src/main/java/org/keycloak/protocol/docker/installation/DockerQuickstartInstallationProvider.java

```diff
@@ -0,0 +1,153 @@
+                                         final String realmName, final String clientName) throws IOException {
+        final DockerComposeZipContent zipContent = new DockerComposeZipContent(realmCert, realmBaseURl, realmName, clientName);
+
+        zipOutput.putNextEntry(new ZipEntry(QUICKSTART_ROOT_DIR));
+
+        // Write docker compose file
+        zipOutput.putNextEntry(new ZipEntry(QUICKSTART_ROOT_DIR + "docker-compose.yaml"));
+        zipOutput.write(zipContent.getYamlFile().generateDockerComposeFileBytes());
+        zipOutput.closeEntry();
+
+        // Write data directory
+        zipOutput.putNextEntry(new ZipEntry(QUICKSTART_ROOT_DIR + zipContent.getDataDirectoryName() + "/"));
+        zipOutput.putNextEntry(new ZipEntry(QUICKSTART_ROOT_DIR + zipContent.getDataDirectoryName() + "/.gitignore"));
+        zipOutput.write("*".getBytes());
+        zipOutput.closeEntry();
+
+        // Write certificates
+        final String certsDirectory = QUICKSTART_ROOT_DIR + zipContent.getCertsDirectory().getDirectoryName() + "/";
+        zipOutput.putNextEntry(new ZipEntry(certsDirectory));
+        zipOutput.putNextEntry(new ZipEntry(certsDirectory + zipContent.getCertsDirectory().getLocalhostCertFile().getKey()));
+        zipOutput.write(zipContent.getCertsDirectory().getLocalhostCertFile().getValue());
+        zipOutput.closeEntry();
+        zipOutput.putNextEntry(new ZipEntry(certsDirectory + zipContent.getCertsDirectory().getLocalhostKeyFile().getKey()));
+        zipOutput.write(zipContent.getCertsDirectory().getLocalhostKeyFile().getValue());
+        zipOutput.closeEntry();
+        zipOutput.putNextEntry(new ZipEntry(certsDirectory + zipContent.getCertsDirectory().getIdpTrustChainFile().getKey()));
+        zipOutput.write(zipContent.getCertsDirectory().getIdpTrustChainFile().getValue());
+        zipOutput.closeEntry();
+
+        // Write README to .zip
+//        try {
+//            final Path path = Paths.get(ClassLoader.getSystemResource("DockerQuickstartReadme.md").toString());
+            zipOutput.putNextEntry(new ZipEntry(QUICKSTART_ROOT_DIR + "README.md"));
+            zipOutput.write(DockerQuickstartInstallationProvider.class.getResourceAsStream("/DockerQuickstartReadme.md").toString().getBytes());
+            zipOutput.closeEntry();
+//        } catch (URISyntaxException e) {
+//            log.warn("Could not find Docker Quickstart Readme file with classloader.  Zip archive will omit README instructions", e);
+//        }
+
+        zipOutput.close();
+        byteStream.close();
+
+        return Response.ok(byteStream.toByteArray(), getMediaType()).build();
+    }
+
+    @Override
+    public String getProtocol() {
+        return DockerAuthV2Protocol.LOGIN_PROTOCOL;
+    }
+
+    @Override
+    public String getDisplayType() {
+        return "Quickstart";
+    }
+
```

Comment:
I think it would make more sense to call this 'Docker Machine YAML' rather than 'Quickstart'. There is a [Keycloak Quickstarts repository](https://github.com/keycloak/keycloak-quickstarts), and it would make more sense to create a submodule 'docker' there and move all the quickstart related documentation there.